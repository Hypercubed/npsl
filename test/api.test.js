/* global test, expect, describe */
import {resolve, join} from 'path';
import {runScript} from '../';

const fixturesPath = resolve(__dirname, './fixtures');

describe('javascript', () => {
  const configPath = join(fixturesPath, 'scripts.js');
  test('default task', () => {
    return runScript({path: configPath}).then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('simple string task', () => {
    return runScript({path: configPath}, 'test').then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('task default', () => {
    return runScript({path: configPath}, 'lint').then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('deep task', () => {
    return runScript({path: configPath}, 'l.s.t').then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('array of tasks', () => {
    return runScript({path: configPath}, 'check').then(res => {
      expect(res).toMatchSnapshot();
    });
  });
});

describe('json', () => {
  const configPath = join(fixturesPath, 'scripts.json');
  test('default task', () => {
    return runScript({path: configPath}).then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('simple string task', () => {
    return runScript({path: configPath}, 'test').then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('task default', () => {
    return runScript({path: configPath}, 'lint').then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('deep task', () => {
    return runScript({path: configPath}, 'l.s.t').then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('array of tasks', () => {
    return runScript({path: configPath}, 'check').then(res => {
      expect(res).toMatchSnapshot();
    });
  });
});

describe('yml', () => {
  const configPath = join(fixturesPath, 'scripts.yml');
  test('default task', () => {
    return runScript({path: configPath}).then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('simple string task', () => {
    return runScript({path: configPath}, 'test').then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('task default', () => {
    return runScript({path: configPath}, 'lint').then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('deep task', () => {
    return runScript({path: configPath}, 'l.s.t').then(res => {
      expect(res).toMatchSnapshot();
    });
  });

  test('array of tasks', () => {
    return runScript({path: configPath}, 'check').then(res => {
      expect(res).toMatchSnapshot();
    });
  });
});
