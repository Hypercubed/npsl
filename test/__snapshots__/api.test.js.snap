exports[`javascript array of tasks 1`] = `
Array [
  Object {
    "subtasks": Array [
      Object {
        "script": "echo test script",
        "stderr": "",
        "stdout": "test script",
        "task": [Function],
        "title": "echo test script",
      },
      Object {
        "script": "echo lint.default",
        "stderr": "",
        "stdout": "lint.default",
        "task": [Function],
        "title": "echo lint.default",
      },
    ],
    "task": [Function],
    "title": "check",
  },
]
`;

exports[`javascript deep task 1`] = `
Array [
  Object {
    "script": "echo deeply nested thing",
    "stderr": "",
    "stdout": "deeply nested thing",
    "task": [Function],
    "title": "l.s.t",
  },
]
`;

exports[`javascript default task 1`] = `
Array [
  Object {
    "script": "echo default script",
    "stderr": "",
    "stdout": "default script",
    "task": [Function],
    "title": "default",
  },
]
`;

exports[`javascript function task 1`] = `
Array [
  Object {
    "task": [Function],
    "title": "fun",
  },
]
`;

exports[`javascript simple string task 1`] = `
Array [
  Object {
    "script": "echo test script",
    "stderr": "",
    "stdout": "test script",
    "task": [Function],
    "title": "test",
  },
]
`;

exports[`javascript task default 1`] = `
Array [
  Object {
    "script": "echo lint.default",
    "stderr": "",
    "stdout": "lint.default",
    "task": [Function],
    "title": "lint",
  },
]
`;

exports[`javascript task with title 1`] = `
Array [
  Object {
    "script": "echo Hello World",
    "stderr": "",
    "stdout": "Hello World",
    "task": [Function],
    "title": "Hello World",
  },
]
`;

exports[`json array of tasks 1`] = `
Array [
  Object {
    "subtasks": Array [
      Object {
        "script": "echo test script",
        "stderr": "",
        "stdout": "test script",
        "task": [Function],
        "title": "echo test script",
      },
      Object {
        "script": "echo lint.default",
        "stderr": "",
        "stdout": "lint.default",
        "task": [Function],
        "title": "echo lint.default",
      },
    ],
    "task": [Function],
    "title": "check",
  },
]
`;

exports[`json deep task 1`] = `
Array [
  Object {
    "script": "echo deeply nested thing",
    "stderr": "",
    "stdout": "deeply nested thing",
    "task": [Function],
    "title": "l.s.t",
  },
]
`;

exports[`json default task 1`] = `
Array [
  Object {
    "script": "echo default script",
    "stderr": "",
    "stdout": "default script",
    "task": [Function],
    "title": "default",
  },
]
`;

exports[`json simple string task 1`] = `
Array [
  Object {
    "script": "echo test script",
    "stderr": "",
    "stdout": "test script",
    "task": [Function],
    "title": "test",
  },
]
`;

exports[`json task default 1`] = `
Array [
  Object {
    "script": "echo lint.default",
    "stderr": "",
    "stdout": "lint.default",
    "task": [Function],
    "title": "lint",
  },
]
`;

exports[`json task with title 1`] = `
Array [
  Object {
    "script": "echo Hello World",
    "stderr": "",
    "stdout": "Hello World",
    "task": [Function],
    "title": "Hello World",
  },
]
`;

exports[`yml array of tasks 1`] = `
Array [
  Object {
    "subtasks": Array [
      Object {
        "script": "echo test script",
        "stderr": "",
        "stdout": "test script",
        "task": [Function],
        "title": "echo test script",
      },
      Object {
        "script": "echo lint.default",
        "stderr": "",
        "stdout": "lint.default",
        "task": [Function],
        "title": "echo lint.default",
      },
    ],
    "task": [Function],
    "title": "check",
  },
]
`;

exports[`yml deep task 1`] = `
Array [
  Object {
    "script": "echo deeply nested thing",
    "stderr": "",
    "stdout": "deeply nested thing",
    "task": [Function],
    "title": "l.s.t",
  },
]
`;

exports[`yml default task 1`] = `
Array [
  Object {
    "script": "echo default script",
    "stderr": "",
    "stdout": "default script",
    "task": [Function],
    "title": "default",
  },
]
`;

exports[`yml simple string task 1`] = `
Array [
  Object {
    "script": "echo test script",
    "stderr": "",
    "stdout": "test script",
    "task": [Function],
    "title": "test",
  },
]
`;

exports[`yml task default 1`] = `
Array [
  Object {
    "script": "echo lint.default",
    "stderr": "",
    "stdout": "lint.default",
    "task": [Function],
    "title": "lint",
  },
]
`;

exports[`yml task with title 1`] = `
Array [
  Object {
    "script": "echo Hello World",
    "stderr": "",
    "stdout": "Hello World",
    "task": [Function],
    "title": "Hello World",
  },
]
`;
